ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java

```diff
@@ -318,6 +352,87 @@ private void tryClearChangeFiles(List<DataFile> changeFiles) {
+              e -> new MixedFileEntry(e.getFile(), e.getTsBound(), true));
+      CloseableIterable<MixedFileEntry> baseEntries =
+          CloseableIterable.transform(
+              fileScan(baseTable, dataFilter, expirationConfig),
+              e -> new MixedFileEntry(e.getFile(), e.getTsBound(), false));
+      IcebergTableMaintainer.ExpireFiles changeExpiredFiles =
+          new IcebergTableMaintainer.ExpireFiles();
+      IcebergTableMaintainer.ExpireFiles baseExpiredFiles =
+          new IcebergTableMaintainer.ExpireFiles();
+
+      try (CloseableIterable<MixedFileEntry> entries =
+          CloseableIterable.withNoopClose(
+              com.google.common.collect.Iterables.concat(changeEntries, baseEntries))) {
+        Queue<MixedFileEntry> fileEntries = new LinkedTransferQueue<>();
+        entries.forEach(
+            e -> {
+              if (mayExpired(e, partitionFreshness, expireTimestamp)) {
+                fileEntries.add(e);
+              }
+            });
+        fileEntries
+            .parallelStream()
+            .filter(e -> willNotRetain(e, expirationConfig, partitionFreshness))
+            .forEach(
+                e -> {
+                  if (e.isChange()) {
+                    changeExpiredFiles.addFile(e);
+                  } else {
+                    baseExpiredFiles.addFile(e);
+                  }
+                });
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+
+      return Lists.newArrayList(changeExpiredFiles, baseExpiredFiles);
+    }
+
+    @Override
+    public List<IcebergTableMaintainer.ExpireFiles> expiredFileScan(
+        DataExpirationConfig expirationConfig,
+        Expression dataFilter,
+        long expireTimestamp,
+        Map<StructLike, IcebergTableMaintainer.DataFileFreshness> partitionFreshness) {
+      return arcticTable.isKeyedTable()
+          ? keyedExpiredFileScan(expirationConfig, dataFilter, expireTimestamp, partitionFreshness)
+          : expiredFileScan(expirationConfig, dataFilter, expireTimestamp, partitionFreshness);
+    }
+
+    @Override
+    public void doExpireFiles(
+        List<IcebergTableMaintainer.ExpireFiles> expiredFiles, long expireTimestamp) {
+      AtomicInteger index = new AtomicInteger();
+      Optional.ofNullable(changeMaintainer)
+          .ifPresent(
+              c ->
+                  c.expireFiles(
+                      IcebergTableUtil.getSnapshotId(getTable(), false),
+                      expiredFiles.get(index.getAndIncrement()),
+                      expireTimestamp));
+      Optional.ofNullable(baseMaintainer)
+          .ifPresent(
+              c ->
+                  c.expireFiles(
+                      IcebergTableUtil.getSnapshotId(getTable(), false),
+                      expiredFiles.get(index.get()),
+                      expireTimestamp));
+    }
   }
 
   public class BaseTableMaintainer extends IcebergTableMaintainer {
```

Comment:
It seems that the base is currently directly inheriting from Iceberg, without considering whether the files in the change are correct or not.